pyiron_atomistics/atomistics/structure/neighbors.py

```diff
@@ -1047,3 +1047,147 @@ def get_cluster(dist_vec, ind_vec, prec=prec):
             ind_shell.append(ia_shells_dict)
         return ind_shell
 
+
+class NeighborsTraj:
+
+    """
+    This class generates the neighbors for a given atomistic trajectory. The assumption here is that the trajectory is
+    canonical (no change in the number and types of species throughout the trajectory). The resulting indices,
+    distances, and vectors are stored as numpy arrays.
+
+    """
+
+    def __init__(self, init_structure=None, positions=None, cells=None, num_neighbors=12, **kwargs):
+        """
+
+        Args:
+            init_structure (pyiron_atomistics.atomistics.structure.atoms.Atoms): Any given structure of the trajectory
+            positions (numpy.ndarray): The cartesian positions of the trajectories
+            cells (numpy.ndarray/None): The varying cell shapes
+            num_neighbors (int): The cutoff for the number of neighbors
+            **kwargs (dict): Additional arguments to be passed to the `get_neighbors()` routine
+                             (eg. cutoff_radius, norm_order , etc.)
+        """
+        self._init_structure = init_structure
+        self._neighbor_indices = None
+        self._neighbor_distances = None
+        self._neighbor_vectors = None
+        self._positions = positions
+        self._cells = cells
+        self._num_neighbors = num_neighbors
+        self._get_neighbors_kwargs = kwargs
```

Comment:
For `__init__` to be compatible a change like this would be necessary.  When `init` is given the other parameters are not set, because presumably they are set in the `init` read from HDF5.  If you want to be complete, you can check that `init` really contains all the attributes that `__init__` normally sets and set those manually that are not in `init.

```suggestion
    def __init__(self, init=None, init_structure=None, positions=None, cells=None, num_neighbors=12, **kwargs):
        """

        Args:
            init_structure (pyiron_atomistics.atomistics.structure.atoms.Atoms): Any given structure of the trajectory
            positions (numpy.ndarray): The cartesian positions of the trajectories
            cells (numpy.ndarray/None): The varying cell shapes
            num_neighbors (int): The cutoff for the number of neighbors
            **kwargs (dict): Additional arguments to be passed to the `get_neighbors()` routine
                             (eg. cutoff_radius, norm_order , etc.)
        """
        super().__init__(init=init)
        if init is None:
            self._init_structure = init_structure
            self._neighbor_indices = None
            self._neighbor_distances = None
            self._neighbor_vectors = None
            self._positions = positions
            self._cells = cells
            self._num_neighbors = num_neighbors
            self._get_neighbors_kwargs = kwargs
```